coding-styles/css/README.md

```diff
@@ -0,0 +1,430 @@
++ We use **<abbr title="Block Element Module">BEM</abbr>** class names and organise our projects with **<abbr title="Inverted Triangle CSS">ITCSS</abbr>** in mind.
+
+
+## Formatting
+
++ Use two space indents, not tabs.
++ Use dashes, not camelCasing in class names. Underscores are only OK for BEM Modules `.main-nav__item {}` (see [BEM and OOCSS](#bem-and-oocss) below).
++ Do not use ID selectors `#hash-is-bad {}`.
++ Put spaces after `:` in property declarations `color:∙#f00;`.
++ Put spaces before `{` in rule declarations `.selector∙{}`.
++ Put line breaks between rulesets.
++ Grouped selectors should be on their own single line.
++ Place the `}` closing braces of declaration blocks on a new line.
++ Each declaration should appear on its own line for more accurate error reporting.
++ End all declarations with a semi-colon `;`.
++ Comma-separated property values should include a space after each comma `box-shadow: 0 1px 2px #ccc, inset 0 1px 0 #fff;`.
++ Don't include spaces after commas within: `rgb()`, `rgba()`, `hsl()`, `hsla()`, or `rect()` values. This helps differentiate multiple color values (comma, no space) from multiple property values (comma with space) `background-color: rgba(0,0,0,.5);`.
++ Don't prefix property values or color parameters with a leading zero: `.5` instead of `0.5` and `-.5px` instead of `-0.5px`.
++ Don't include units for zero values: `margin: 0;` instead of `margin: 0px;`.
++ Lowercase all hex values: `#fff`. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
++ Use shorthand hex values where available, `#fff` instead of `#ffffff`.
++ Quote attribute values in selectors, using double quotes `input[type="text"]`.
++ Never reference `.js-` prefixed class names in CSS. `.js-` are used **exclusively in JavaScript** files.
+
+##### Bad
+```scss
+.avatar{
+  border-radius:50%;
+  border:2px solid white; }
+.no, .nope, .not_good {
+  // ...
+}
+#oi-definitely-no-IDs {
+  // ...
+}
+.js-prefix-not-in-css {
+  // ...
+}
+```
+
+##### Good
+```scss
+.avatar {
+  border-radius: 50%;
+  border: 2px solid #fff;
+}
+
+.one,
+.selector,
+.per-line {
+  // ...
+}
+```
+
+
+## Declaration Order
+Related property declarations should be grouped together following the order:
+
+1. Positioning
+2. Box model
+3. Typographic
+4. Visual
+
+Positioning comes first because it can remove an element from the normal flow of the document and override box model related styles. The box model comes next as it dictates a component's dimensions and placement.
+
+Everything else takes place inside the component or without impacting the previous two sections, and thus they come last.
+
+```scss
+.declaration-order {
+  // Positioning
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 100;
+
+  // Box-model
+  display: block;
+  float: right;
+  width: 100px;
+  height: 100px;
+
+  // Typography
+  font: normal 13px "Helvetica Neue", sans-serif;
```

Comment:
can you put which rule this has violated?  
maybe this?
`Don't include units for zero values: margin: 0; instead of margin: 0px;.` this is only for 0 value